### PR TITLE
Rebase: Add include_lazy_artifacts flag to create_app (issues/464).

### PR DESCRIPTION
**Pull request by KristofferC**
cc @Sacha0, does this look ok to you?

### CODE REVIEW

File: src/PackageCompiler.jl [hunk 1]
```diff
@@ -760,7 +763,8 @@ function create_app(package_dir::String,
                     audit=true,
                     force=false,
                     c_driver_program::String=joinpath(@__DIR__, "embedding_wrapper.c"),
-                    cpu_target::String=default_app_cpu_target())
+                    cpu_target::String=default_app_cpu_target(),
+                    include_lazy_artifacts::Bool=true)
```
Comment (Sacha0) [inline src/PackageCompiler.jl:767]: I imagine I'm missing something, but at first blush it seems that this argument should be passed to the `_create_app` call in this method's body? :)

File: src/PackageCompiler.jl [hunk 2]
```diff
@@ -851,6 +855,9 @@ compiler.
    Used to determine and set the `compatibility_version` (on Apple) and `soname` (on
    Linux/UNIX).
 
+- `include_lazy_artifacts::Bool`: if lazy artifacts should be included in the bundled artifats,
```
Comment (Sacha0) [inline src/PackageCompiler.jl:858]: `artifats` 😄

File: src/PackageCompiler.jl [hunk 3]
```diff
@@ -746,6 +746,9 @@ compiler.
 
 - `force::Bool`: Remove the folder `compiled_app` if it exists before creating the app.
 
+- `include_lazy_artifacts::Bool`: if lazy artifacts should be included in the bundled artifats,
```
Comment (Sacha0) [inline src/PackageCompiler.jl:749]: `artifats` here too 😄

**Comment by Sacha0** [comment]
The only potentially important comment is the the first, in case I wasn't missing something there :).

File: src/PackageCompiler.jl (hunk 1 above)
Comment (kmsquire) [reply src/PackageCompiler.jl:767]: I don't think you're missing anything--it definitely should be passed.

MR: https://github.com/JuliaLang/PackageCompiler.jl/pull/530

**Pull request merged**